ibis/backends/flink/__init__.py

```diff
@@ -442,3 +444,67 @@ def _get_operations(cls):
     @classmethod
     def has_operation(cls, operation: type[ops.Value]) -> bool:
         return operation in cls._get_operations()
+
+    def insert(
+        self,
+        table_name: str,
+        obj: pd.DataFrame | ir.Table | list | dict,
+        database: str | None = None,
+        catalog: str | None = None,
+        overwrite: bool = False,
+    ) -> TableResult:
```

Comment:
Ibis itself is already inconsistent about whether `insert` calls return anything.  The other sqlalchemy backends return `None` -- Impala and PySpark both return some `result` object. 
Is there useful information encoded in the `TableResult`?  If not, I'd say we can just execute and not return it.

Reply:
When `TableResult` is returned, we can call `wait()` on the `TableResult` to wait for the job to finish. I think it's at least useful for debugging purposes for now, but I don't feel strongly about returning it, although `_exec_sql` also just returns the `TableResult`.

Reply:
That seems like a good reason to return it!